File: app/src/main/java/cl/figonzal/lastquakechile/CrashlyticsTree.java
package cl.figonzal.lastquakechile;

import android.util.Log;

import com.google.firebase.crashlytics.FirebaseCrashlytics;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import timber.log.Timber;

public class CrashlyticsTree extends Timber.Tree {


    @SuppressWarnings("UnnecessaryReturnStatement")
    @Override
    protected void log(int priority, @Nullable String tag, @NotNull String message, @Nullable Throwable t) {

        if (priority == Log.VERBOSE || priority == Log.DEBUG || priority == Log.INFO) {
            return;
        } else {

            FirebaseCrashlytics.getInstance().log(message);
            if (t != null) {
                FirebaseCrashlytics.getInstance().recordException(t);
            }
        }
    }
}

File: app/src/main/java/cl/figonzal/lastquakechile/CrashlyticsTree.kt
package cl.figonzal.lastquakechile

import android.util.Log
import com.google.firebase.crashlytics.FirebaseCrashlytics
import timber.log.Timber

class CrashlyticsTree : Timber.Tree() {

    override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {

        when (priority) {
            Log.VERBOSE, Log.DEBUG, Log.INFO -> return
            else -> {
                FirebaseCrashlytics.getInstance().log(message)
                if (t != null) {
                    FirebaseCrashlytics.getInstance().recordException(t)
                }
            }
        }
    }
}